jenkins_jobs/modules/hipchat_notif.py

```diff
@@ -131,12 +133,26 @@ def gen_xml(self, xml_parent, data):
                 logger.warning("'room' is deprecated, please use 'rooms'")
                 hipchat['rooms'] = [hipchat['room']]
 
-        properties = xml_parent.find('properties')
-        if properties is None:
-            properties = XML.SubElement(xml_parent, 'properties')
-        pdefhip = XML.SubElement(properties,
-                                 'jenkins.plugins.hipchat.'
-                                 'HipChatNotifier_-HipChatJobProperty')
+        plugin_info = self.registry.get_plugin_info("Jenkins HipChat Plugin")
+        version = pkg_resources.parse_version(plugin_info.get('version', '0'))
+
+        if version >= pkg_resources.parse_version("0.1.9"):
+            publishers = xml_parent.find('publishers')
+            if publishers is None:
+                publishers = XML.SubElement(xml_parent, 'publishers')
+
+            logger.warning(
+                "'hipchat' module supports the old plugin versions <1.9, "
+                "newer versions are supported via the 'publishers' module. "
+                "Please upgrade you job definition")
+            return hipchat_pub(self.registry, publishers, hipchat)
```

Comment:
I'm not calling for updating bunch of other files to latest master. However, I've find the base commit between us and upstream to be d5a60242a9fbc6b9c87ac93887ac1a69f88939dc (`git merge-base gooddata/gd-patches openstack-infra/master`)

Then when you diff the file in this PR HEAD (3ba14a7ed453387c097e49851a3a45e5c3d1c381) vs. in that base commit, you get (among others):
```diff
git diff d5a60242a9fbc6b9c87ac93887ac1a69f88939dc 3ba14a7ed453387c097e49851a3a45e5c3d1c381 -- jenkins_jobs/modules/hipchat_notif.py
--- a/jenkins_jobs/modules/hipchat_notif.py
+++ b/jenkins_jobs/modules/hipchat_notif.py
@@ -144,7 +145,7 @@ class HipChat(jenkins_jobs.modules.base.Base):
                 "'hipchat' module supports the old plugin versions <1.9, "
                 "newer versions are supported via the 'publishers' module. "
                 "Please upgrade you job definition")
-            return self.registry.dispatch('publisher', publishers, data)
+            return hipchat_pub(self.registry, publishers, hipchat)
         else:
             properties = xml_parent.find('properties')
             if properties is None:
```

So are you definitely sure that we shouldn't use `self.registry.dispatch`? I couldn't tell, because the tests are failing in both cases :)